File: Python/LongestPalindromicSubstring.py
def longestPalindromicSubstring(s):

    n = len(s)
    dp = [[False for i in range(n)] for j in range(n)]

    for i in range(n):
        dp[i][i] = True

    maxLen = 1
    result = s[0]

    for i in range(n-1):
        if s[i] == s[i+1]:
            dp[i][i+1] = True
            maxLen = 2
            result = s[i:i+2]

    for k in range(3,n+1):
        for i in range(n-k+1):
            j = i+k-1
            if dp[i+1][j-1] and s[i] == s[j]:
                dp[i][j] = True
                if k>maxLen:
                    maxLen = k
                    result = s[i:i+k] 

    return result

File: Javascript/LongestPalindromicSubstring.js
var longestPalindromicSubstring = function(s){
    const n = s.length
    const dp = []

    for (let i=0;i<n;i++){
        dp[i] = new Array(n).fill(false)
    }

    for (let i=0;i<n;i++){
        dp[i][i] = true
    }

    let result = s[0]
    let maxLen = 1

    for (let i=0;i<n-1;i++){
        if (s[i] === s[i+1]){
            dp[i][i+1] = true
            maxLen = 2
            result = s.slice(i,i+2)
        }
    }

    for (let k=3;k<n+1;k++){
        for(let i=0;i<n-k+1;i++){
            let j = i+k-1
            if (dp[i+1][j-1] && (s[i] === s[j])){
                dp[i][j] = true
                if (k>maxLen){
                    maxLen = k
                    result = s.slice(i,i+k)
                }
            }
        }
    }

    return result
}